server/collectionPub/permissions.js

```diff
@@ -20,3 +21,58 @@ export const getPermissions = async ({ userId, communityId, collectionId }) => {
 		destroy: isAuthenticated,
 	};
 };
+
+const canManagePub = async ({ userId, communityId, pubId }) => {
+	const scopeData = await getScope({ loginId: userId, communityId: communityId, pubId: pubId });
+	return (
+		scopeData.activePermissions.canManage &&
+		scopeData.elements.activePub.communityId === communityId
+	);
+};
+
+export const canCreateCollectionPub = async ({ userId, communityId, collectionId }) => {
+	const { activePermissions } = await getScope({
+		communityId: communityId,
+		collectionId: collectionId,
+		loginId: userId,
+	});
+	return activePermissions.canManage;
+};
+
+export const getUpdatableFieldsForCollectionPub = async ({
+	userId,
+	communityId,
+	collectionPubId,
+}) => {
+	const { pubId, collectionId } = await CollectionPub.findOne({ where: { id: collectionPubId } });
+	const {
+		elements: { activeCollection },
+		activePermissions,
+	} = await getScope({
+		communityId: communityId,
+		collectionId: collectionId,
+		loginId: userId,
+	});
+	if (activePermissions.canManage) {
+		return ['rank', 'pubRank', 'contextHint'];
+	}
+	const canUpdatePubRank = await canManagePub({
+		userId: userId,
+		communityId: activeCollection.communityId,
+		pubId: pubId,
+	});
+	if (canUpdatePubRank) {
+		return ['pubRank'];
+	}
+	return null;
+};
+
+export const canDestroyCollectionPub = async ({ userId, communityId, collectionPubId }) => {
+	const { collectionId } = await CollectionPub.findOne({ where: { id: collectionPubId } });
+	const { activePermissions } = await getScope({
+		communityId: communityId,
+		collectionId: collectionId,
+		loginId: userId,
+	});
+	return activePermissions.canManage;
+};
```

Comment:
I have a sneaking suspicion this code may have triggered your comment about object shorthand syntax...